app/models/api_service.rb

```diff
@@ -2,12 +2,13 @@ class APIService < ApplicationRecord
   URL_OPEN_AI = "https://api.openai.com/"
   URL_ANTHROPIC = "https://api.anthropic.com/"
   URL_GROQ = "https://api.groq.com/openai/v1/"
+  URL_GEMINI = "https://api.gemini.com/v1/"
```

Comment:
Is this the correct URL? When I was doing a curl test I was using: `https://generativelanguage.googleapis.com/` but this is definitely the URL you're using locally?

Let's add a migration so that everyone gets this new API service automatically created for them. I think the migration would just have an up and look like this (assuming this URL_GEMINI is correct):

**This is what I did, but I'm hoping this is actually mistaken and it's the reason why mine isn't working.**

```
    User.all.each do |user|
      user.api_services.create!(
        name: "Google Gemini",
        driver: "gemini",
        url: "https://api.gemini.com/v1/"
      )
    end
```

Be sure to run this migration in your dev environment and commit the schema change. And then also edit the `models.yml` file and add this to the top of it, assuming all these details are correct:

```
- api_name: gemini-1.5-pro-002
  name: Google Gemini 1.5 Pro (latest)
  supports_images: true
  supports_tools: false
  input_token_cost_cents: '0'
  output_token_cost_cents: '0'
  best: true
  supports_system_message: true
  api_service_name: Google Gemini
  position: 0
```

These two changes will ensure it gets added for everyone when we merge this PR in.

Reply:
I don't remember where I got that URL from but it's not the correct one. However, the gem does not get the API URL from any source. It just receives the input from the service and builds the URL inside. In this case the service is used: "generative-language-api" with version: "v1beta" so the gem uses this URL inside: "https://generativelanguage.googleapis.com/v1beta/". And if the "vertex-ai-api" service is chosen it makes a more complicated configuration with credentials that are not taken into account right now.